### ISSUE TITLE
Should we keep the possibility of witness txn to contain the output which has an asset that Alice transfers to Bob?

### ISSUE THREAD

**Issue by UkolovaOlga** [LNP-BP/LNPBPs#32]
**Witness transaction** - a txn that contains the commitment; is a part of single use seal witness.

### Intro
RGB uses bitcoin transactions in 2 ways: 
1st - to put commitments to what happens with RGB off-chain (commit to each transfer which is also called 'state transition') into the bitcoin transaction.
2nd - to use UTXO to allocate some state to, in particular to allocate some assets to this output. And when this output is spent, it means that Alice actually did a transfer of that asset and the txn that spends this output, must contain the commitment (meaning it should be a witness txn). 

Basically Alice is using 2 bitcoin transactions: one she needs to assign asset to, other - the witness transaction. Key idea from the early days was to combine these 2 transactions into 1, meaning that the witness transaction may contain the output which contains the asset that Alice is transferring to Bob, meaning that Alice assigns the asset not to some other existing transaction output, but to the output that will be created by Alice.  

### The question is if we need to keep this opportunity?

Raised during the dev call on June 24th, 2020

**Comment by dr-orlovsky** [comment]
Decision: we need it (see discussion during the last call)
